src/webgpu/shader/execution/expression/call/builtin/atomics/atomicStore.spec.ts

```diff
@@ -18,6 +28,73 @@ fn atomicStore(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T)
 `
   )
   .params(u =>
-    u.combine('SC', ['storage', 'uniform'] as const).combine('T', ['i32', 'u32'] as const)
+    u
+      .combine('workgroupSize', workgroupSizes)
+      .combine('dispatchSize', dispatchSizes)
+      .combine('mapId', keysOf(kMapId))
+      .combine('scalarType', ['u32', 'i32'])
   )
-  .unimplemented();
+  .fn(t => {
+    const numInvocations = t.params.workgroupSize * t.params.dispatchSize;
+    const bufferNumElements = numInvocations;
+    const mapId = kMapId[t.params.mapId];
+    const extra = mapId.wgsl(numInvocations, t.params.scalarType); // Defines map_id()
+
+    const initValue = 0;
+    const op = `atomicStore(&output[id], map_id(id))`;
```

Comment:
Similar comment here. Nothing really atomic about this, when there's a 1:1 invocation to memory location.
Suggest multiple invocations writing to the same location with different values, then checking that the written value was one of the values. That would at least demonstrate there was no tearing in the written word.

Reply:
Did as you suggested. Added two more atomicStore "_advanced" tests that read/write to the same atomic value.